### DISCUSSION TITLE
Is there a way to traverse Groups, starting at the root

### DISCUSSION THREAD

**Discussion by josiahslack** [BlueBrain/HighFive#693] [Questions] [answered]
I've been experimenting with HighFive over the past few days, and have mostly been able to figure out how to do what I wanted. However, today I needed the capability of opening an HDF5 file and walking through the tree of Groups. I dug around the examples and the source, but didn't find anything promising. What am I missing?

**Comment by 1uc** [comment, marked as answer]
You can try one of these:
https://github.com/BlueBrain/HighFive/blob/master/include/highfive/bits/H5Node_traits.hpp#L135
https://github.com/BlueBrain/HighFive/blob/master/include/highfive/bits/H5Node_traits.hpp#L130

Or
https://github.com/BlueBrain/HighFive/blob/master/include/highfive/bits/H5Node_traits.hpp#L158

Which gives you control over the order in which you'd access the object. Not all `IndexType`s are available by default, depending on how the file was created. The order can sometimes matter a lot in terms of performance. However, as with anything performance related, first establish if you need to worry about performance, then measure if it's working.

This will tell you what a certain object is:
https://github.com/BlueBrain/HighFive/blob/master/include/highfive/bits/H5Node_traits.hpp#L179

**Reply by josiahslack** [reply]
Thanks! A combination of `getGroup()` and `listObjectNames()` was just what I needed.
